HTTP doesn't seem to work using windows

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

HTTP segment on windows doesn't seem to work. The same config on linux (WSL) works fine.

### Theme

```yaml
- type: rprompt
      alignment: right
      segments:
        - type: http
          foreground: "#ffffff"
          background: "#000000"
          style: diamond
          leading_diamond: 
          invert_powerline: true
          template: " {{ .Body.ip }} "
          options:
            url: https://api.ipify.org?format=json
            method: GET
          cache:
            strategy: device
            duration: 1h
```

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
Version: 29.10.0

Shell: shell (7.6.1)

Prompt:

    ~                                                                                                                                                                    R.E.M. - The One I Love - Remastered 2012  06:45:37 
shell ❯

Segments:

ConsoleTitle(true)  -   0 ms
Os(true)            -   0 ms
Session(false)      -   0 ms
Session(false)      -   0 ms
Path(true)          -   1 ms
Git(false)          -   0 ms
Root(false)         -   0 ms
Executiontime(false) -   0 ms
Status(true)        -   0 ms
Spotify(true)       -   1 ms
Time(true)          -   0 ms
Text(true)          -   0 ms
Status(true)        -   0 ms
Http(false)         - 10000 ms

Run duration: 10.006303s

Cache path: C:\Users\<user>\AppData\Local\Packages\ohmyposh.cli_96v55e8n804z4\LocalCache\Local\oh-my-posh

Config path: C:\Users\<user>\.ohmyposh.yaml

Logs:
<useless lines skipped>
[DEBUG] 06:45:37.246 segment.go:Execute:173 → segment: Http
[DEBUG] 06:45:37.246 store.go:]:176 → (session) found entry: toggle_cache - map[]
[TRACE] 06:45:37.246 store.go:](session toggle_cache) - 0s
[DEBUG] 06:45:37.246 segment.go:isToggled:291 → no toggles found
[DEBUG] 06:45:37.246 store.go:]:163 → (device) key not found: segment_cache_Http
[TRACE] 06:45:37.246 store.go:](device segment_cache_Http) - 0s
[DEBUG] 06:45:37.246 segment.go:restoreCache:316 → no cache found for segment: Http, key: segment_cache_Http
[TRACE] 06:45:37.246 jobs_windows.go:CreateJobForGoroutine(creating job for goroutine(Http): 55) - 0s
[DEBUG] 06:45:37.246 map.go:String:83 → url: https://api.ipify.org/?format=json
[DEBUG] 06:45:37.246 map.go:String:83 → method: GET
[DEBUG] 06:45:37.246 terminal.go:HTTPRequest:403 ↓
    GET /?format=json HTTP/1.1
    Host: api.ipify.org
    User-Agent: Go-http-client/1.1
    Accept-Encoding: gzip


[ERROR] 06:45:47.247 terminal.go:HTTPRequest:408 → Get "https://api.ipify.org/?format=json": context deadline exceeded
[TRACE] 06:45:47.247 terminal.go:HTTPRequest(https://api.ipify.org/?format=json) - 10.0005623s
[TRACE] 06:45:47.247 terminal.go:Flags() - 0s
[TRACE] 06:45:47.247 terminal.go:Flags() - 0s
```

### Expected behaviour (from ubuntu WSL)
```
[TRACE] 06:44:23.343 terminal.go:Flags() - 20ns
[TRACE] 06:44:23.343 terminal.go:Shell() - 51ns
[TRACE] 06:44:23.343 terminal.go:Flags() - 20ns
[TRACE] 06:44:23.343 terminal.go:Flags() - 70ns
[DEBUG] 06:44:23.343 home.go:12 → /home/kkprince
[TRACE] 06:44:23.343 terminal.go:GOOS() - 30ns
[DEBUG] 06:44:23.343 segment.go:Execute:173 → segment: Http
[DEBUG] 06:44:23.343 store.go:]:176 → (session) found entry: toggle_cache - map[]
[TRACE] 06:44:23.343 store.go:](session toggle_cache) - 3.257µs
[DEBUG] 06:44:23.343 segment.go:isToggled:291 → no toggles found
[DEBUG] 06:44:23.343 map.go:String:83 → url: https://api.ipify.org?format=json
[DEBUG] 06:44:23.343 map.go:String:83 → method: GET
[DEBUG] 06:44:23.343 terminal.go:HTTPRequest:403 ↓
    GET /?format=json HTTP/1.1
    Host: api.ipify.org
    User-Agent: Go-http-client/1.1
    Accept-Encoding: gzip
    
    
[DEBUG] 06:44:23.449 terminal.go:HTTPRequest:429 → {"ip":"0.0.0.0"}
[TRACE] 06:44:23.449 terminal.go:HTTPRequest(https://api.ipify.org?format=json) - 106.027591ms
[TRACE] 06:44:23.449 terminal.go:Flags() - 161ns
[DEBUG] 06:44:23.449 text.go:patchTemplate:149 →  {{ .Data.Body.ip }} 
[TRACE] 06:44:23.449 terminal.go:Shell() - 120ns
[TRACE] 06:44:23.449 terminal.go:Flags() - 20ns
```

--- a/src/segments/http.go
+++ b/src/segments/http.go
@@ -16,8 +16,7 @@ type HTTP struct {
 }
 
 const (
-	METHOD  options.Option = "method"
-	TIMEOUT options.Option = "timeout"
+	METHOD options.Option = "method"
 )
 
 func (h *HTTP) Template() string {
@@ -31,7 +30,7 @@ func (h *HTTP) Enabled() bool {
 	}
 
 	method := h.options.String(METHOD, "GET")
-	timeout := h.options.Int(TIMEOUT, 10000)
+	timeout := h.options.Int(options.HTTPTimeout, 10000)
 
 	if resolved, err := template.Render(url, nil); err == nil {
 		url = resolved

--- a/src/segments/http.go
+++ b/src/segments/http.go
@@ -16,7 +16,8 @@ type HTTP struct {
 }
 
 const (
-	METHOD options.Option = "method"
+	METHOD  options.Option = "method"
+	TIMEOUT options.Option = "timeout"
 )
 
 func (h *HTTP) Template() string {
@@ -30,12 +31,13 @@ func (h *HTTP) Enabled() bool {
 	}
 
 	method := h.options.String(METHOD, "GET")
+	timeout := h.options.Int(TIMEOUT, 10000)
 
 	if resolved, err := template.Render(url, nil); err == nil {
 		url = resolved
 	}
 
-	result, err := h.getResult(url, method)
+	result, err := h.getResult(url, method, timeout)
 	if err != nil {
 		return false
 	}
@@ -44,12 +46,12 @@ func (h *HTTP) Enabled() bool {
 	return true
 }
 
-func (h *HTTP) getResult(url, method string) (map[string]any, error) {
+func (h *HTTP) getResult(url, method string, timeout int) (map[string]any, error) {
 	setMethod := func(request *http.Request) {
 		request.Method = method
 	}
 
-	resultBody, err := h.env.HTTPRequest(url, nil, 10000, setMethod)
+	resultBody, err := h.env.HTTPRequest(url, nil, timeout, setMethod)
 	if err != nil {
 		return nil, err
 	}

--- a/src/segments/http_test.go
+++ b/src/segments/http_test.go
@@ -73,7 +73,7 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 			}
 
 			if tc.timeout > 0 {
-				props[TIMEOUT] = tc.timeout
+				props[options.HTTPTimeout] = tc.timeout
 			}
 
 			env.On("HTTPRequest", tc.url).Return([]byte(tc.response), func() error {

--- a/src/segments/http_test.go
+++ b/src/segments/http_test.go
@@ -17,6 +17,7 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 		name        string
 		url         string
 		method      string
+		timeout     int
 		response    string
 		shouldError bool
 	}{
@@ -24,6 +25,7 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 			name:        "Valid URL with GET response",
 			url:         "https://jsonplaceholder.typicode.com/posts/1",
 			method:      "GET",
+			timeout:     0,
 			response:    `{"id": "1"}`,
 			expected:    "1",
 			shouldError: false,
@@ -32,6 +34,7 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 			name:        "Valid URL with POST response",
 			url:         "https://jsonplaceholder.typicode.com/posts",
 			method:      "POST",
+			timeout:     0,
 			response:    `{"id": "101"}`,
 			expected:    "101",
 			shouldError: false,
@@ -40,12 +43,23 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 			name:        "Valid URL with error response",
 			url:         "https://api.example.com/data",
 			method:      "GET",
+			timeout:     0,
 			shouldError: true,
 		},
 		{
 			name:        "Empty URL",
 			url:         "",
 			method:      "GET",
+			timeout:     0,
+			shouldError: false,
+		},
+		{
+			name:        "Custom timeout",
+			url:         "https://jsonplaceholder.typicode.com/posts/1",
+			method:      "GET",
+			timeout:     5000,
+			response:    `{"id": "2"}`,
+			expected:    "2",
 			shouldError: false,
 		},
 	}
@@ -56,6 +70,10 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 			props := options.Map{
 				URL:    tc.url,
 				METHOD: tc.method,
+			}
+
+			if tc.timeout > 0 {
+				props[TIMEOUT] = tc.timeout
 			}
 
 			env.On("HTTPRequest", tc.url).Return([]byte(tc.response), func() error {

--- a/src/segments/http_test.go
+++ b/src/segments/http_test.go
@@ -3,13 +3,27 @@ package segments
 import (
 	"encoding/json"
 	"errors"
+	"io"
 	"testing"
 
+	runtimehttp "github.com/jandedobbeleer/oh-my-posh/src/runtime/http"
 	"github.com/jandedobbeleer/oh-my-posh/src/runtime/mock"
 	"github.com/jandedobbeleer/oh-my-posh/src/segments/options"
 
 	"github.com/stretchr/testify/assert"
 )
+
+// timeoutCapturingEnv wraps the mock environment to record the timeout argument
+// passed to HTTPRequest, so tests can assert it flows through correctly.
+type timeoutCapturingEnv struct {
+	*mock.Environment
+	capturedTimeout int
+}
+
+func (e *timeoutCapturingEnv) HTTPRequest(url string, body io.Reader, timeout int, modifiers ...runtimehttp.RequestModifier) ([]byte, error) {
+	e.capturedTimeout = timeout
+	return e.Environment.HTTPRequest(url, body, timeout, modifiers...)
+}
 
 func TestHTTPSegmentEnabled(t *testing.T) {
 	cases := []struct {
@@ -66,7 +80,7 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
-			env := new(mock.Environment)
+			inner := new(mock.Environment)
 			props := options.Map{
 				URL:    tc.url,
 				METHOD: tc.method,
@@ -76,22 +90,27 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 				props[options.HTTPTimeout] = tc.timeout
 			}
 
-			env.On("HTTPRequest", tc.url).Return([]byte(tc.response), func() error {
+			inner.On("HTTPRequest", tc.url).Return([]byte(tc.response), func() error {
 				if tc.shouldError {
 					return errors.New("error")
 				}
 				return nil
 			}())
 
+			capturing := &timeoutCapturingEnv{Environment: inner}
+
 			cs := &HTTP{
 				Base: Base{
-					env:     env,
+					env:     capturing,
 					options: props,
 				},
 			}
 
 			_ = cs.Enabled()
 			assert.Equal(t, tc.expected, cs.Body["id"], tc.name)
+			if tc.timeout > 0 {
+				assert.Equal(t, tc.timeout, capturing.capturedTimeout, "expected custom timeout to be passed to HTTPRequest")
+			}
 		})
 	}
 }

--- a/src/segments/http_test.go
+++ b/src/segments/http_test.go
@@ -17,8 +17,8 @@ func TestHTTPSegmentEnabled(t *testing.T) {
 		name        string
 		url         string
 		method      string
-		timeout     int
 		response    string
+		timeout     int
 		shouldError bool
 	}{
 		{

--- a/website/docs/segments/web/http.mdx
+++ b/website/docs/segments/web/http.mdx
@@ -30,10 +30,11 @@ import Config from "@site/src/components/Config.js";
 
 ## Options
 
-| Name     |   Type   | Default | Description                                         |
-| -------- | :------: | :-----: | --------------------------------------------------- |
-| `url`    | `string` |   ``    | The HTTP URL you want to call, supports [templates] |
-| `method` | `string` |  `GET`  | The HTTP method to use, `GET` or `POST`             |
+| Name      |   Type   |  Default  | Description                                         |
+| --------- | :------: | :-------: | --------------------------------------------------- |
+| `url`     | `string` |    ``     | The HTTP URL you want to call, supports [templates] |
+| `method`  | `string` |   `GET`   | The HTTP method to use, `GET` or `POST`             |
+| `timeout` |  `int`   | `10000`   | The timeout in milliseconds for the HTTP request    |
 
 ## Template ([info][templates])
 

--- a/website/docs/segments/web/http.mdx
+++ b/website/docs/segments/web/http.mdx
@@ -34,7 +34,7 @@ import Config from "@site/src/components/Config.js";
 | --------- | :------: | :-------: | --------------------------------------------------- |
 | `url`     | `string` |    ``     | The HTTP URL you want to call, supports [templates] |
 | `method`  | `string` |   `GET`   | The HTTP method to use, `GET` or `POST`             |
-| `timeout` |  `int`   | `10000`   | The timeout in milliseconds for the HTTP request    |
+| `http_timeout` |  `int`   | `10000`   | The timeout in milliseconds for the HTTP request    |
 
 ## Template ([info][templates])
 

